app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java

```diff
@@ -178,15 +165,14 @@ public void onPageSelected(int position) {
                         tabLayout.getTabAt(CONTRIBUTIONS_TAB_POSITION).select();
                         isContributionsFragmentVisible = true;
                         updateMenuItem();
-
                         break;
                     case NEARBY_TAB_POSITION:
                         Timber.d("Nearby tab selected");
                         tabLayout.getTabAt(NEARBY_TAB_POSITION).select();
                         isContributionsFragmentVisible = false;
                         updateMenuItem();
                         // Do all permission and GPS related tasks on tab selected, not on create
-                        ((NearbyFragment)contributionsActivityPagerAdapter.getItem(1)).onTabSelected(onOrientationChanged);
+                        ((NearbyParentFragment)contributionsActivityPagerAdapter.getItem(1)).nearbyParentFragmentPresenter.onTabSelected();
```

Comment:
Why is this referring to child fragment's presenter. We should ideally just let the fragment take care of it, this should just inform the fragment that the tab is selected. Also, is there a way the fragment could determine by itself that it is visible and in resumed state, if that is what we are trying to achieve.

Reply:
Presenter.onTabSelected does not just decides to fragment state, it adds location listener, gets permissions if needed, checks broadcast receiver. So this method actually behaves like a controller method.